e3sm_supported_machines/create_new_e3sm_unified_env.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python
+import subprocess
+import os
+import socket
+import glob
+import stat
+import pwd
+import grp
+import requests
+import progressbar
+
+
+def check_env(base_path, env_name):
+    print("Checking the environment {}".format(env_name))
+    env = os.environ
+    env['CDAT_ANONYMOUS_LOG'] = 'no'
+
+    activate = 'source {}/etc/profile.d/conda.sh; conda activate {}'.format(
+        base_path, env_name)
+
+    imports = ['vcs', 'ILAMB', 'acme_diags', 'mpas_analysis', 'livvkit',
+               'IPython', 'globus_cli', 'zstash']
+    for import_name in imports:
+        command = '{}; python -c "import {}"'.format(activate, import_name)
+        try:
+            subprocess.check_call(command, env=env, executable='/bin/bash',
+                                  shell=True)
+        except subprocess.CalledProcessError as e:
+            print('  {} failed'.format(import_name))
+            raise e
+        print('  {} passes'.format(import_name))
+
```

Comment:
This command subprocesses and check block is repeated 3 times so it might be worth abstracting it into it's own function

Reply:
A pleasant surprise that anyone is paying attention!

I did move these to a function as you recommended.  They were initially trivial calls but became more complex as time went on...